docs/clients/promtail/scraping.md

```diff
@@ -0,0 +1,211 @@
+# Promtail Scraping (Service Discovery)
+
+## File Target Discovery
+
+Promtail discovers locations of log files and extract labels from them through
+the `scrape_configs` section in the config YAML. The syntax is identical to what
+[Prometheus uses](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#scrape_config).
+
+`scrape_configs` contains one or more entries which are executed for each
+discovered target (i.e., each container in each new pod running in the
+instance):
+
+```
+scrape_configs:
+  - job_name: local
+    static_configs:
+      - ...
+
+  - job_name: kubernetes
+    kubernetes_sd_config:
+      - ...
+```
+
+If more than one scrape config section matches your logs, you will get duplicate
+entires as lthe logs are sent in different streams likely with slightly
+different labels.
+
+There are different types of labels present in Promtail:
+
+* Labels starting with `__` (two underscores) are internal labels. They usually
+  come from dynamic sources like service discovery. Once relabeling is done,
+  they are removed from the label set. To persist internal labels so they're
+  sent to Loki, rename them so they don't start with `__`.
+
+* Labels starting with `__meta_kubernetes_pod_label_*` are "meta labels" which
+  are generated based on your Kubernetes pod's labels.
+
+  For exmaple, if your Kubernetes pod has a label `name` set to `foobar`, then
+  the `scrape_configs` section will recieve an internal label
+  `__meta_kubernetes_pod_label_name` with a value set to `foobar`.
+
+* Other labels starting with `__meta_kubernetes_*` exist based on other
+  Kubernetes metadata,s such as the namespace of the pod
+  (`__meta_kubernetes_namespace`) or the name of the container inside the pod
+  (`__meta_kubernetes_pod_container_name`). Refer to
+  [the Prometheus docs](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#kubernetes_sd_config)
+  for the full list of Kubernetes meta labels.
+
+* The `__path__` label is a special label which Promtail uses after discovery to
+  figure out where the file to read is located. Wildcards are allowed.
+
+* The label `filename` is added for every file found in `__path__` to ensure the
+  uniqueness of the streams. It is set to the absolute path of the file the line
+  was read from.
+
+### Kubernetes Discovery
+
+Note that while promtail can utilize the Kubernetes API to discover pods as
+targets, it can only use pods that are running on the same hostname of the
+machine that promtail is running on.
```

Comment:
on the same hostname = is that the same node?

Reply:
Yeah, we validate it's the same node by using the hostname. I'll expand